src/components/mainpage/timeline/TimelineItem.js

```diff
@@ -0,0 +1,173 @@
+      marginLeft: theme.spacing(4),
+      marginBottom: theme.spacing(2),
+      width: "100%",
+      display: "flex",
+    },
+  },
+  icon: {
+    marginRight: "0.5rem",
+    cursor: "pointer",
+  },
+  btn: {
+    marginRight: theme.spacing(4),
+  },
+}));
+
+const TimelineItem = ({ id, photographer, src, like, comments }) => {
+  const classes = useStyles();
+  const { dispatch } = usePhotosContext();
+  const [newComment, setNewComment] = useState("");
+  const [moreComments, setMoreComments] = useState(false);
+
+  const onAddlike = (id) => {
+    const updates = {
+      like: {
+        isLiked: true,
+        numOfLikes: like.numOfLikes + 1,
+      },
+    };
+    dispatch({
+      type: "EDIT_PHOTO",
+      payload: {
+        id,
+        updates,
+      },
+    });
+    localStorage.setItem(`L-${id}`, JSON.stringify(updates.like));
+  };
+
+  const onRemoveLike = () => {
+    const updates = {
+      like: {
+        isLiked: false,
+        numOfLikes: like.numOfLikes - 1,
+      },
+    };
+    dispatch({
+      type: "EDIT_PHOTO",
+      payload: {
+        id,
+        updates,
+      },
+    });
+    localStorage.removeItem(`L-${id}`);
+  };
+
+  const onAddNewComment = (id) => {
+    setNewComment("");
+    const commentInfo = {
+      user: "user",
+      commentUid: uuid(),
+      sentence: newComment,
+    };
+    comments.unshift(commentInfo);
+    const updates = {
+      comments,
+    };
+    dispatch({ type: "EDIT_PHOTO", payload: { id, updates } });
+    localStorage.setItem(`C-${id}`, JSON.stringify(comments));
+  };
+
+  const showMoreComments = () => {
+    setMoreComments(true);
+  };
+
+  const showLessComments = () => {
+    setMoreComments(false);
+  };
+
+  return (
+    <div className="item-wrap">
+      <img src={src.large} alt={photographer} />
+      <p className="photographer">Photographer: {photographer}</p>
+      <div className="icon-wrap">
+        <div className="fav-icon-wrap">
+          {like.isLiked ? (
+            <FavoriteIcon
+              className={classes.icon}
+              onClick={() => onRemoveLike(id)}
+              style={{ color: "red" }}
+            />
+          ) : (
+            <FavoriteBorderIcon
+              className={classes.icon}
+              onClick={() => onAddlike(id)}
+            />
+          )}
+          <p className="num-of-likes">{like.isLiked ? like.numOfLikes : ""}</p>
+        </div>
+        <ChatBubbleOutlineOutlinedIcon className={classes.icon} />
+        <SendOutlinedIcon className={classes.icon} />
+        <BookmarkBorderOutlinedIcon className={classes.icon} />
+      </div>
+      <div className="comments-display">
+        {comments.length !== 0 ? (
+          comments.map((comment, index) => (
+            <p
+              style={
+                index >= 1 && !moreComments
+                  ? { display: "none" }
+                  : { display: "initial" }
+              }
+              key={comment.commentUid}
+            >
```

Comment:
Maybe you should rewrite without setting style (display: none & display: initial). Because React can have conditions inside JSX.